web/client-ide/src/main/java/io/deephaven/ide/client/IdeConnection.java

```diff
@@ -3,15 +3,32 @@
 import elemental2.promise.Promise;
 import io.deephaven.web.client.api.QueryConnectable;
 import io.deephaven.web.shared.fu.JsRunnable;
-import io.deephaven.web.shared.ide.ConsoleAddress;
+import io.deephaven.web.shared.data.ConnectToken;
+import jsinterop.annotations.JsConstructor;
 import jsinterop.annotations.JsIgnore;
+import jsinterop.annotations.JsMethod;
+import jsinterop.annotations.JsPackage;
 import jsinterop.annotations.JsType;
+import java.nio.charset.StandardCharsets;
 
 /**
  */
 @JsType(namespace = "dh")
 public class IdeConnection extends QueryConnectable<IdeConnection> {
-    private final ConsoleAddress address;
+    @JsMethod(namespace = JsPackage.GLOBAL)
+    private static native String atob(String encodedData);
+
+    private static AuthTokenPromiseSupplier getAuthTokenPromiseSupplier(IdeConnectionOptions options) {
+        ConnectToken token = null;
+        if (options != null && options.getAuthToken() != null) {
+            token = new ConnectToken();
+            token.setBytes(atob(options.getAuthToken()).getBytes(StandardCharsets.ISO_8859_1));
```

Comment:
this can probably die... or even switch to utf8, it is specific to enterprise at this time

Reply:
Switched to UTF8